src/codemagic/tools/_app_store_connect/actions/publish_action.py

```diff
@@ -394,6 +409,18 @@ def _process_ipa_after_upload(
                 'max_build_processing_wait': 0,  # Overwrite waiting since we already waited above.
                 'version_string': app_store_options.version_string or ipa.version,
             }
+            if app_store_options.cancel_previous_submissions:
+                states_to_cancel = (
+                    ReviewSubmissionState.WAITING_FOR_REVIEW,
+                    ReviewSubmissionState.IN_REVIEW,
+                    ReviewSubmissionState.UNRESOLVED_ISSUES,
```

Comment:
Is it really necessary to cancel all those states? Wouldn't `states_to_cancel = (ReviewSubmissionState.WAITING_FOR_REVIEW,)` suffice?